### PR TITLE
app-dicts/wordnet: -r6 updates db files to 3.1

### PR DESCRIPTION
**Pull request by zyklotomic**
Updated the database files for WordNet from 3.0 to 3.1

### CODE REVIEW

**Comment by zyklotomic** [comment]
I think my proxy-maint request on bugzilla might be relevant: 
https://bugs.gentoo.org/show_bug.cgi?id=615006

File: app-dicts/wordnet/wordnet-3.0-r6.ebuild [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	"${WORKDIR}"/${P}-CVE-2008-2149.patch #211491
+
+	"${FILESDIR}"/${P}-tcl8.6.patch
+	"${FILESDIR}"/${P}-format-security.patch
+	"${FILESDIR}"/${P}-src_stubs_c.patch
+	"${FILESDIR}"/${P}-fix-indexing-bug-314799.patch
+)
+
+src_prepare() {
+	epatch "${PATCHES[@]}"
```
Comment (mgorny) [inline app-dicts/wordnet/wordnet-3.0-r6.ebuild]: We'd prefer eapply here. Do the patches fail at consistent -p?

File: app-dicts/wordnet/wordnet-3.0-r6.ebuild [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	fi
+
+	# Drop installation of OLD tk.h headers #255590
+	sed '/^SUBDIRS/d' -i include/Makefile.am || die
+	sed 's: include/tk/Makefile::' -i configure.ac || die
+	rm -rf include/tk/ || die
+
+	rm -f configure || die
+	eautoreconf
+	MAKEOPTS+=" -e"
```
Comment (mgorny) [inline app-dicts/wordnet/wordnet-3.0-r6.ebuild]: Please do not alter MAKEOPTS. Instead, pass `-e` to emake (yeah, you'd need to redefine src_compile, src_install and possible src_test if there are any tests).

Furthermore, I don't think `-e` is a good option to use. It'd be better to figure out what needs overrides, and pass it explicitly, e.g. `emake CFLAGS="${CFLAGS}" ...`

File: app-dicts/wordnet/wordnet-3.0-r6.ebuild [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+	econf \
+		--with-tcl="${EPREFIX}"/usr/$(get_libdir) \
+		--with-tk="${EPREFIX}"/usr/$(get_libdir)
+}
+
+pkg_preinst(){
+	# For clarification, WN is still on  version 3.0. Only the database files 
+	# have been updated to 3.1 as a package for 3.1 does not currently exist.
+	rm -r "${D}"/usr/share/wordnet/dict
+	mv "${WORKDIR}"/dict "${D}"/usr/share/wordnet
```
Comment (mgorny) [inline app-dicts/wordnet/wordnet-3.0-r6.ebuild]: 1. This all belongs in src_install, not _preinst. _preinst is run after the package is ready, and doesn't (shouldn't) have access to WORKDIR.
2. Both need `||die`

File: app-dicts/wordnet/wordnet-3.0-r6.ebuild (hunk 1 above)
Comment (zyklotomic) [inline app-dicts/wordnet/wordnet-3.0-r6.ebuild]: Yes, there is partial success with -p1. Have problem with applying wordnet-3.0-CVE-2008-2149.patch. That's where the ebuild stops.

File: app-dicts/wordnet/wordnet-3.0-r6.ebuild [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	WNHOME="${EPREFIX}/usr/share/wordnet" \
+	econf \
+		--with-tcl="${EPREFIX}"/usr/$(get_libdir) \
+		--with-tk="${EPREFIX}"/usr/$(get_libdir)
+}
+
+src_install(){
+	# For clarification, WN is still on  version 3.0. Only the database files
+	# have been updated to 3.1 as a package for 3.1 does not currently exist.
+	emake -e DESTDIR="${D}" install
```
Comment (mgorny) [inline app-dicts/wordnet/wordnet-3.0-r6.ebuild]: You probably/especially also want `emake -e` in `src_compile()`.

File: app-dicts/wordnet/metadata.xml [hunk 5]
```diff
@@ -1,5 +1,12 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
 <pkgmetadata>
-<!-- maintainer-needed -->
+	<maintainer type="person">
+			<email>chocopuff298+gentoo@gmail.com</email>
```
Comment (mgorny) [inline app-dicts/wordnet/metadata.xml]: As the script says, you need to use an e-mail that corresponds to a Gentoo Bugzilla account.

Comment (SoapGentoo) [inline app-dicts/wordnet/metadata.xml]: (also, incorrect indentation)

Comment (zyklotomic) [inline app-dicts/wordnet/metadata.xml]: Fixed on bugzilla end by changing my bugzilla email.

File: app-dicts/wordnet/metadata.xml [hunk 6]
```diff
@@ -1,5 +1,12 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
 <pkgmetadata>
-<!-- maintainer-needed -->
+<maintainer type="person">
+	<email>chocopuff298+gentoo@gmail.com</email>
+	<name> Ethan Kiang</name>
```
Comment (mgorny) [inline app-dicts/wordnet/metadata.xml]: Extra whitespace. I'll fix it when merging.

**Pull request closed without merging**